graph/paths_counts_naive.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "adjacency_matrix.h"
+
+template<typename GraphT>
+AdjacencyMatrix<std::uint32_t> count_paths_of_len(const GraphT& graph, size_t /*path_len*/) {
```

Comment:
Lets declare the next function in the header
`template<typename GraphT>`
`GraphT count_paths_of_len(const GraphT& graph, size_t path_len);`

So an output type is the same as an input type.

And then define implementations in the impl file using [partial template specialization](https://www.fluentcpp.com/2017/08/15/function-templates-partial-specialization-cpp/)

Reply:
Had to make declaration more complex, since we count number of paths (some integer type) while input graph may have different types of edges